### ISSUE TITLE
Gtk4 docs?

### ISSUE THREAD

**Issue by tschoonj** [lazka/pgi-docs#3]
I was wondering if it would be possible to add docs for Gtk4. Is this something I can do myself through a pull request?

I would like to start migrating a pygobject project from Gtk 3 to 4 and having your docs up and running for that would make my life a lot easier 😄 

Thanks!

**Comment by lazka** [comment]
It's a bit of a bad timing. It builds from Debian testing/unstable, but that doesn't have gtk4 and Debian is in freeze right now.

**Comment by tschoonj** [comment]
Ok no problem. I see that you have some CI configured: does this mean that you use a cron job to update the docs regularly based on changes in debian:unstable?

**Comment by lazka** [comment]
yeah, in theory yes. it's currently broken though due to some broken gir in Debian. Likely I need to block some packages.

edit: fixed now

**Comment by keithbowes** [comment]
Yeah, right now, how to do things in GTK 4 is heavily buried (and the C API can't be simply mapped to Python).  The primary example was creating a builder scope worked in a quick C example I did, but in PyGObject, it wouldn't send the parameters to the callbacks and would crash on the second call back.  I randomly bumped into the solution in the PyGobject git history: you send the signals to map in the Gtk.Builder constructor itself.

So instead of:

```
builder = Gtk.Builder()
scope = Gtk.BuilderCScope()
scope.add_callback_symbol("about", about)
scope.add_callback_symbol("exit", exit")
builder.set_scope(scope)
builder.add_from_file("application.ui")
```

You do:

```
builder = Gtk.Builder({"about": about, "exit": exit})
builder.add_from_from("application.ui")
```

It's unfortunate that for compatibility sake, Gtk.Builder.connect_signals() couldn't have stayed instead of the new constructor, but that really is impossible:  GTK 3 requires the signals to be connected after loading the builder file, where GTK 4 requires it to be done before.

Though, it's not only PyGObject's GTK 4 support that's badly documented.  I've run into problems upgrading my theme too, as it seems to be badly documented.  E.g. popup menus supposedly have an element content with the class .background, but that didn't work; instead, the popup has the class .background and content doesn't have any class.  I figured a lot of it out by looking at how Adwaita does it, as the docs were wrong about several things.  Maybe when GTK 4 matures a bit, things will get better.
